griptape/drivers/prompt_model/bedrock_claude_prompt_model_driver.py

```diff
@@ -1,23 +1,41 @@
 import json
-from attr import define, field, Factory
+from attr import define, field
 from griptape.artifacts import TextArtifact
 from griptape.utils import PromptStack
-from griptape.drivers import BasePromptModelDriver
+from griptape.drivers import BasePromptModelDriver, AmazonBedrockPromptDriver
 from griptape.tokenizers import BedrockClaudeTokenizer
 
 
 @define
 class BedrockClaudePromptModelDriver(BasePromptModelDriver):
-    model: str = field(kw_only=True)
-    tokenizer: BedrockClaudeTokenizer = field(
-        default=Factory(
-            lambda self: BedrockClaudeTokenizer(model=self.model),
-            takes_self=True
-        ),
-        kw_only=True,
-    )
     top_p: float = field(default=0.999, kw_only=True)
     top_k: int = field(default=250, kw_only=True)
+    _tokenizer: BedrockClaudeTokenizer = field(default=None, kw_only=True)
+
+    @property
+    def tokenizer(self) -> BedrockClaudeTokenizer:
+        """Returns the tokenizer for this driver.
+
+        We need to pass the `session` field from the Prompt Driver to the
+        Tokenizer. However, the Prompt Driver is not initialized until after
+        the Prompt Model Driver is initialized. To resolve this, we make the `tokenizer`
+        field a @property that is only initialized when it is first accessed.
+        This ensures that by the time we need to initialize the Tokenizer, the 
+        Prompt Driver has already been initialized.
+
+        See this thread more more information: https://github.com/griptape-ai/griptape/issues/244
+
+        Returns:
+            BedrockClaudeTokenizer: The tokenizer for this driver.
+        """
+        if self._tokenizer:
+            return self._tokenizer
+        else:
+            if isinstance(self.prompt_driver, AmazonBedrockPromptDriver):
```

Comment:
To make it more explicit, let's override the prompt driver in this class with:

```python
prompt_driver: Optional[AmazonBedrockPromptDriver] = field(default=None, kw_only=True)
```

And let's drop the type check here. We are not doing it anywhere else in the framework, as we are relying on type hints for that.